### PR TITLE
update project

### LINKED ISSUE #6: update gradle files

### PR DESCRIPTION
**Pull request by jperedadnr**
Update project files (Gradle files, scripts)

Fixes #6

### CODE REVIEW

File: gradle/android_project/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-bin.zip
```
Comment (nlisker) [inline gradle/android_project/gradle/wrapper/gradle-wrapper.properties:5]: Any reason to use this version and not 8.7?

File: gradle/android_project/build.gradle [hunk 2]
```diff
@@ -2,17 +2,15 @@ buildscript {
     repositories {
         mavenCentral()
         google()
-        jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.0-alpha09'
+        classpath 'com.android.tools.build:gradle:8.1.1'
```
Comment (nlisker) [inline gradle/android_project/build.gradle:7]: Any reason to use this version and not 8.3.1?

File: gradle/native-build.gradle [hunk 3]
```diff
@@ -123,7 +123,7 @@ ext.aarBuild = { buildDir, projectDir, name, os ->
     if (sdk == null) {
         throw new GradleException("Error: ANDROID_SDK can't be null")
     }
-    def androidJar = sdk + "/platforms/android-29/android.jar"
+    def androidJar = sdk + "/platforms/android-33/android.jar"
```
Comment (nlisker) [inline gradle/native-build.gradle:126]: I suggest moving this version string out, like in Attach.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-bin.zip
```
Comment (nlisker) [inline gradle/wrapper/gradle-wrapper.properties:3]: Same question about 8.7.

File: gradle/native-build.gradle
Comment (nlisker) [inline gradle/native-build.gradle:1]: Line 363: `def platform="aarch64-linux-android21"`

Is the `21` here corresponding to the minSdkVersion?

**Review by nlisker — COMMENTED** [review]
During the build I get these warnings:
```
> Task :library:compileDebugJavaWithJavac
warning: [options] source value 8 is obsolete and will be removed in a future release

warning: [options] target value 8 is obsolete and will be removed in a future release
> Task :library:mergeDebugGeneratedProguardFiles
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
```

And this error:
```
FAILURE: Build failed with an exception.
> Task :library:bundleDebugAar
* What went wrong:finished
Execution failed for task ':log:androidBuild'.
> A problem occurred starting process 'command '/home/nlisker/.gluon/substrate/Android/ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/clang''
> Task :log:androidBuild FAILEDd
```
I checked the path above and there's a directory missing in it denoting the version:
/home/nlisker/.gluon/substrate/Android/ndk/25.2.9519653/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
This was created by GluonFX, I didn't download the ndk manually.

Also note that in the documentation, the ndk path is named `ndk-bundle`, but I think it's only named that if it's installed from Android Studio while GluonFX just uses `ndk`.

File: gradle/android_project/build.gradle (hunk 2 above)
Comment (jperedadnr) [reply gradle/android_project/build.gradle]: fixed

File: gradle/android_project/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (jperedadnr) [reply gradle/android_project/gradle/wrapper/gradle-wrapper.properties]: fixed

File: gradle/native-build.gradle (hunk 3 above)
Comment (jperedadnr) [reply gradle/native-build.gradle]: done

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (jperedadnr) [reply gradle/wrapper/gradle-wrapper.properties]: fixed

File: gradle/native-build.gradle
Comment (jperedadnr) [reply gradle/native-build.gradle:1]: I guess it is. But it is consistent with the current setup.

Comment (nlisker) [reply gradle/native-build.gradle:1]: My concern is that if someone wants to change the minSdkVersion it won't be noticeable, so if that's the case, I suggest pulling that version string out. There are already several files where the minSdkVersion is defined.

**Review by johanvos — APPROVED** [review]

**Pull request merged**